evalml/tests/automl_tests/test_automl.py

```diff
@@ -1268,16 +1270,20 @@ def test_max_batches_works(mock_pipeline_fit, mock_score, max_batches, X_y_binar
         # if they are not searched over. That is why n_automl_pipelines does not equal
         # n_results when max_iterations and max_batches are None
         n_automl_pipelines = 1 + len(automl.allowed_pipelines)
+        num_ensemble_batches = 0
     else:
+        # automl algorithm does not know about the additional stacked ensemble pipelines
+        num_ensemble_batches = (max_batches - 1) // ensemble_nth_batch
         # So that the test does not break when new estimator classes are added
-        n_results = 1 + len(automl.allowed_pipelines) + (5 * (max_batches - 1))
+        n_results = 1 + len(automl.allowed_pipelines) + (5 * (max_batches - 1 - num_ensemble_batches)) + num_ensemble_batches
         n_automl_pipelines = n_results
-
     assert automl._automl_algorithm.batch_number == max_batches
-    # We add 1 to pipeline_number because _automl_algorithm does not know about the baseline
     assert automl._automl_algorithm.pipeline_number + 1 == n_automl_pipelines
     assert len(automl.results["pipeline_results"]) == n_results
-    assert automl.rankings.shape[0] == min(1 + len(automl.allowed_pipelines), n_results)
+    if num_ensemble_batches == 0:
+        assert automl.rankings.shape[0] == min(1 + len(automl.allowed_pipelines), n_results)  # add one for baseline
+    else:
+        assert automl.rankings.shape[0] == min(2 + len(automl.allowed_pipelines), n_results)  # add two for baseline and stacked ensemble
```

Comment:
Do we have a check somewhere that asserts stacking has been run? I.e. look at the search order, grab the last pipeline, and assert its a stacked ensembler?

Reply:
Ah I see we do that in the iterative algo tests, nice